coresimd/ppsv/api/swap_bytes.rs

```diff
@@ -42,68 +46,112 @@ macro_rules! impl_swap_bytes {
     };
 }
 
-#[cfg(test)]
-macro_rules! test_swap_bytes {
-    (u8x2) => {
-        #[test]
-        fn swap_bytes() {
-            use coresimd::simd::u8x2;
-            use std::mem;
-
-            let x: u16 = 0x2d99;
-            let expected = x.swap_bytes();
-
-            let vec: u8x2 = unsafe { mem::transmute(x) };
-            let actual = unsafe { mem::transmute(vec.swap_bytes()) };
+macro_rules! impl_test {
+    ($id:ident, $vec8:ident) => {
```

Comment:
I think it would be better to just use a `[u8; 64]`, initialize that, and then use the `$id::{load,store}_unaligned` methods to load that into a vector, call `swap_bytes`/`to_le`/`to_be`, and then write the result back to an `[u8; 64]` array. Then you just compare the first `size_of::<$id>()` bytes of the array against the result that they should have.